contracts/oracle/CurveOracle.sol

```diff
@@ -13,26 +14,48 @@ contract CurveOracle is IBaseOracle {
   using SafeMath for uint;
 
   IBaseOracle public tokenOracle;
+  ICurveRegistry public registry;
 
-  constructor(IBaseOracle _tokenOracle) public {
+  mapping(address => address[]) public ulTokens; // lpToken -> underlying token array
+  mapping(address => address) public poolOf; // lpToken -> pool
+
+  constructor(IBaseOracle _tokenOracle, ICurveRegistry _registry) public {
     tokenOracle = _tokenOracle;
+    registry = _registry;
+  }
+
+  /// @dev Return pool address given LP token and update pool info if not exist.
+  /// @param lp LP token to find the corresponding pool.
+  function getPool(address lp) public returns (address) {
+    address pool = poolOf[lp];
+    if (pool == address(0)) {
+      require(lp != address(0), 'no lp token');
+      pool = registry.get_pool_from_lp_token(lp);
```

Comment:
Will this call revert if the pool does not exist?

Reply:
Default is zero address. Added require.